### PR TITLE
Make GH Action golang static checks more informative on failure

### PR DESCRIPTION
**Pull request by coventry**
This changes the github actions for the golang static checks so that the invocation of the checker tools is more explicit, and results in display of any error messages they produce.

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
-      uses: grandcolline/golang-github-actions@v1.1.0
+    - uses: actions/setup-go@v2
+    - uses: actions/checkout@v2
       with:
-        run: vet
-        comment: false
-  
-  # TODO: Enable linters below one by one and fix issues with each
-
+        path: repo
```
Comment (HenryNguyen5) [inline .github/workflows/static.yml:11]: The default working directory is the checked out repository, we can remove all the `repo` references

File: core/services/vrf/vrf.go [hunk 2]
```diff
@@ -119,6 +119,11 @@ var Generator = secp256k1Curve.Point().Base()
 // HashUint256s returns a uint256 representing the hash of the concatenated byte
 // representations of the inputs
 func HashUint256s(xs ...*big.Int) (*big.Int, error) {
+	eulersCriterionPower := div(sub(fieldSize, one), two)
+	_ = eulersCriterionPower
+	Generator := secp256k1Curve.Point().Base()
+	_ = Generator
+	fmt.Println("%d", "foo")
```
Comment (j16r) [inline core/services/vrf/vrf.go:126]: Looks like a debugging statement?

File: .github/workflows/static.yml (hunk 1 above)
Comment (coventry) [reply .github/workflows/static.yml]: Thanks, @HenryNguyen5 , that cleaned things up a bit.

File: .github/workflows/static.yml [hunk 3]
```diff
@@ -1,43 +1,33 @@
...
-    - uses: actions/checkout@master
-    - name: check
-      uses: grandcolline/golang-github-actions@v1.1.0
-      with:
-        run: imports
-        comment: false
-
+    - uses: actions/setup-go@v2
+    - uses: actions/checkout@v2
+    - run: echo $GOPATH
```
Comment (HenryNguyen5) [inline .github/workflows/static.yml:27]: Debug statement?

File: .github/workflows/static.yml [hunk 4]
```diff
@@ -1,43 +1,33 @@
...
     steps:
-      - uses: actions/checkout@master
-      - name: check
-        uses: grandcolline/golang-github-actions@v1.1.0
-        with:
-          run: shadow
-          comment: false
-
+    - uses: actions/setup-go@v2
+    - uses: actions/checkout@v2
```
Comment (HenryNguyen5) [inline .github/workflows/static.yml:17]: Typically checkouts go first in a workflow

File: .github/workflows/static.yml [hunk 5]
```diff
@@ -59,18 +48,17 @@ jobs:
...
-  #   - uses: actions/checkout@master
-  #   - name: check
-  #     uses: grandcolline/golang-github-actions@v1.1.0
+  #   - uses: actions/setup-go@v2
+  #   - uses: actions/checkout@v2
   #     with:
-  #       run: errcheck
-  #       comment: false
-
+  #       path: repo
```
Comment (HenryNguyen5) [inline .github/workflows/static.yml:58]: Don't need repo refs here

File: .github/workflows/static.yml [hunk 6]
```diff
@@ -59,18 +48,17 @@ jobs:
...
-  #   - name: check
-  #     uses: grandcolline/golang-github-actions@v1.1.0
+  #   - uses: actions/setup-go@v2
+  #   - uses: actions/checkout@v2
   #     with:
-  #       run: errcheck
-  #       comment: false
-
+  #       path: repo      
+  #   - run: echo $GOPATH
```
Comment (HenryNguyen5) [inline .github/workflows/static.yml:59]: Debug statement?

File: .github/workflows/static.yml (hunk 3 above)
Comment (coventry) [reply .github/workflows/static.yml]: Thanks, I removed it.

File: .github/workflows/static.yml (hunk 4 above)
Comment (coventry) [reply .github/workflows/static.yml:16]: I reordered as you suggested.

File: .github/workflows/static.yml (hunk 6 above)
Comment (coventry) [reply .github/workflows/static.yml]: Thanks, removed.

File: core/services/vrf/vrf.go (hunk 2 above)
Comment (coventry) [reply core/services/vrf/vrf.go:126]: That's there to trigger the CI failures. I'll remove the final commit with these probes, before merging.

**Comment by coventry** [comment]
I've removed the failure-probe commit, so this should be ready for approval and merge, now.

File: core/services/vrf/vrf.go (hunk 2 above)
Comment (coventry) [reply core/services/vrf/vrf.go]: I've removed them.

File: .github/workflows/static.yml [hunk 7]
```diff
@@ -59,18 +47,14 @@ jobs:
   #     with:
   #       run: staticcheck
   #       comment: false
-
   # errcheck:
-  #   name: Errcheck
+  #   name: imports
```
Comment (samsondav) [inline .github/workflows/static.yml:51]: Not sure why this name has changed?

Comment (coventry) [reply .github/workflows/static.yml:51]: Nice catch, thanks.

**Comment by coventry** [comment]
> One question: Why only run it on core rather than the whole repo?

In my hands, that was noticeably faster, and I assumed it was because there's a huge tree of irrelevant javascript to trawl. I'll point it at `ingester`, `internal` and `packr`, too.

**Comment by kaleofduty** [comment]
> > One question: Why only run it on core rather than the whole repo?
> 
> In my hands, that was noticeably faster, and I assumed it was because there's a huge tree of irrelevant javascript to trawl. I'll point it at `ingester`, `internal` and `packr`, too.

I'm guessing that there's a whole lot less JS stuff in a fresh git checkout that hasn't been "yarned". (Unless you checked on a fresh clone?) So I'd vote to just run it on everything. If we observe checks taking ages, we can always start restricting scope again.

File: core/adapters/http_test.go [hunk 8]
```diff
@@ -674,6 +674,7 @@ func TestHTTP_BuildingURL(t *testing.T) {
 			ep := adapters.ExtendedPath{}
 			qp := adapters.QueryParameters{}
 			err := json.Unmarshal([]byte(test.path), &ep)
+			assert.NoError(t, err)
```
Comment (coventry) [inline core/adapters/http_test.go:677]: If I wrote a linter, it would say "add a failure message to your `assert`/`require`.

**Comment by coventry** [comment]
Oh, @vlyl, also, prior to merging, would you mind squashing the commits?

**Comment by coventry** [comment]
Thanks, @vlyl. Can you tell us which rule was triggering on the `varname := zeroObject` lines, and what the failure message was?

File: core/adapters/http.go [hunk 9]
```diff
@@ -331,8 +331,8 @@ type QueryParameters url.Values
 
 // UnmarshalJSON implements the Unmarshaler interface
 func (qp *QueryParameters) UnmarshalJSON(input []byte) error {
-	values := url.Values{}
-	strs := []string{}
+	var values = url.Values{}
```
[the remark this answers is not in the document]
Comment (vlyl) [reply core/adapters/http.go]: we can use flag to ignore this check

File: core/adapters/http_test.go (hunk 8 above)
Comment (vlyl) [reply core/adapters/http_test.go]: fix it as you suggestion

**Pull request merged**